pages/hold/request/[id].tsx

```diff
@@ -0,0 +1,340 @@
+                      <>
+                        My Scholar Room
+                        <br />
+                        476 Fifth Avenue (42nd St and Fifth Ave)
+                      </>
+                    }
+                    value="1"
+                  />
+                  <Radio
+                    id="2"
+                    labelText={
+                      <>
+                        Schwarzman Building - Rose Main Reading Room
+                        <br />
+                        315 476 Fifth Avenue (42nd St and Fifth Ave)
+                      </>
+                    }
+                    value="2"
+                  />
+                  <Radio
+                    id="3"
+                    labelText={
+                      <>
+                        Schwarzman Building - Art & Architecture Room 300
+                        <br />
+                        476 Fifth Avenue (42nd St and Fifth Ave)
+                      </>
+                    }
+                    value="3"
+                  />
+                  <Radio
+                    id="4"
+                    labelText={
+                      <>
+                        Schwarzman Building - Dorot Jewish Division Room 111
+                        <br />
+                        476 Fifth Avenue (42nd St and Fifth Ave)
+                      </>
+                    }
+                    value="4"
+                  />
+                </RadioGroup>
+              </FormField>
+              <ButtonGroup>
+                <Button id="holdRequestSubmit" type="submit">
+                  Submit request
+                </Button>
+              </ButtonGroup>
+            </Form>
+          </>
+        )}
+      </Layout>
+    </>
+  )
+}
+
+export async function getServerSideProps({ params, req, res }) {
+  const { id } = params
+
+  // authentication redirect
+  const patronTokenResponse = await initializePatronTokenAuth(req.cookies)
+  const isAuthenticated = patronTokenResponse.isTokenValid
+  const redirectTrackerCookie = req.cookies["nyplAccountRedirects"]
+  const redirectCount = parseInt(redirectTrackerCookie, 10) || 0
+  const redirectBasedOnNyplAccountRedirects =
+    doRedirectBasedOnNyplAccountRedirects(redirectCount)
+
+  // If we end up not authenticated 3 times after redirecting to the login url, don't redirect.
+  if (redirectBasedOnNyplAccountRedirects && !isAuthenticated) {
+    res.setHeader(
+      "Set-Cookie",
+      `nyplAccountRedirects=${
+        redirectCount + 1
+      }; Max-Age=10; path=/; domain=.nypl.org;`
+    )
+    const redirect = getLoginRedirect(req, `/hold/request/[${id}]`)
+
+    return {
+      redirect: {
+        destination: redirect,
+        permanent: false,
+      },
+    }
+  }
+
+  try {
+    const patronId = patronTokenResponse?.decodedPatron?.sub
+
+    // fetch bib and item
+    const [bibId, itemId] = id.split("-")
+
+    if (!itemId) {
+      throw new Error("No item id in url")
```

Comment:
I think at the root, this should be a 404 error and should redirect to that page.

Reply:
It's user error, not an application error.

Reply:
Ah, this is wrapped in a try/catch and does redirect. Great. I commented too early. But for the error on 324, that might not be a 404 but maybe it is a 400?

Reply:
The 404 is just a redirect to the 404 page, we can make the error a 400 in the api route though.